mm2src/mm2_p2p/src/behaviours/atomicdex.rs

```diff
@@ -798,19 +890,29 @@ fn maintain_connection_to_relays(swarm: &mut AtomicDexSwarm, bootstrap_addresses
 
     let mut rng = rand::thread_rng();
     if connected_relays.len() < mesh_n_low {
+        let mut recently_dialed_peers = RECENTLY_DIALED_PEERS.lock().unwrap();
         let to_connect_num = mesh_n - connected_relays.len();
-        let to_connect = swarm
-            .behaviour_mut()
-            .core
-            .peers_exchange
-            .get_random_peers(to_connect_num, |peer| !connected_relays.contains(peer));
+        let to_connect =
+            swarm
+                .behaviour_mut()
+                .core
+                .peers_exchange
+                .get_random_peers(to_connect_num, |peer, addresses| {
+                    addresses
+                        .iter()
+                        .any(|addr| pre_dial_check(&mut recently_dialed_peers, addr))
+                        && !connected_relays.contains(peer)
```

Comment:
we should move the connected_relays check first as there is no need to maker the peer as dialed then find out we are already connected.

also this should be just a check, it shouldn't mutate the recent dialed peers.

Reply:
Makes sense